third_party/sl4a/src/main/java/com/google/android/mobly/snippet/rpc/MethodDescriptor.java

```diff
@@ -267,25 +272,104 @@ String getHelp() {
      */
     public static Object getDefaultValue(Type parameterType, Annotation[] annotations) {
         for (Annotation a : annotations) {
-            if (a instanceof RpcOptional) {
+            if (a instanceof RpcDefault) {
+                RpcDefault defaultAnnotation = (RpcDefault) a;
+                TypeConverter<?> converter =
+                        converterFor(parameterType, defaultAnnotation.converter());
+                return converter.convert(defaultAnnotation.value());
+            } else if (a instanceof RpcOptional) {
                 return null;
             }
         }
         throw new IllegalStateException("No default value for " + parameterType);
     }
 
+    @SuppressWarnings("rawtypes")
+    private static TypeConverter<?> converterFor(
+            Type parameterType, Class<? extends TypeConverter> converterClass) {
+        if (converterClass == TypeConverter.class) {
+            TypeConverter<?> converter = typeConverters.get(parameterType);
+            if (converter == null) {
+                throw new IllegalArgumentException(
+                        String.format("No predefined converter found for %s", parameterType));
+            }
+            return converter;
+        }
+        try {
+            Constructor<?> constructor = converterClass.getConstructor(new Class<?>[0]);
+            return (TypeConverter<?>) constructor.newInstance(new Object[0]);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Cannot create converter from %s", converterClass.getCanonicalName()),
+                    e);
+        }
+    }
+
     /**
      * Determines whether or not this parameter has default value.
      *
      * @param annotations annotations of the parameter
      */
     public static boolean hasDefaultValue(Annotation[] annotations) {
         for (Annotation a : annotations) {
-            if (a instanceof RpcOptional) {
+            if (a instanceof RpcDefault || a instanceof RpcOptional) {
```

Comment:
an optional param may not have a default value right?
is this method name accurate?

Reply:
The Optional param will have a `null` as its default value.

Reply:
These params are special bc they can be optional, and having a default value is just a consequence of that right? Bc users probably care more about these param being optional, not that they can have default values?

Reply:
Got it, separated their method naming. Thanks!